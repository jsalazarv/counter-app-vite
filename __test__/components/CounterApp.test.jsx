import { render, screen, fireEvent } from '@testing-library/react';
import { Counter } from '../../src/components/Counter/Counter.jsx';

describe('[Snapshots]', () => {
  test('Should match snapshot', () => {
    const value = 0;
    const { container } = render(<Counter value={value} />);

    expect(container).toMatchSnapshot();
  });

  test('Should contain an initial value equal to 100', () => {
    const value = 100;
    const { container } = render(<Counter value={value} />);

    expect(screen.getByText(100)).toBeTruthy();
  });
});

describe('[Events]', () => {
  test('Should increment the counter with the +1 button', () => {
    const value = 0;
    render(<Counter value={value} />);
    fireEvent.click(screen.getByRole('button', { name: 'add' }));
    fireEvent.click(screen.getByRole('button', { name: 'add' }));
    fireEvent.click(screen.getByRole('button', { name: 'add' }));

    expect(screen.getByText(3)).toBeTruthy();
  });

  test('Should diminish the counter with the -1 button', () => {
    const value = 0;
    render(<Counter value={value} />);
    fireEvent.click(screen.getByRole('button', { name: 'subtract' }));

    expect(screen.getByText(-1)).toBeTruthy();
  });

  test('Should reset the counter value', () => {
    const value = 0;
    render(<Counter value={value} />);
    fireEvent.click(screen.getByRole('button', { name: 'add' }));
    fireEvent.click(screen.getByRole('button', { name: 'add' }));
    fireEvent.click(screen.getByRole('button', { name: 'add' }));
    fireEvent.click(screen.getByRole('button', { name: 'reload' }));

    expect(screen.getByText(0)).toBeTruthy();
  });
});
