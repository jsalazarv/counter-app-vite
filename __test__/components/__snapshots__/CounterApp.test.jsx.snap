// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Snapshots] Should match snapshot 1`] = `
<div>
  <div
    class="wrapper"
  >
    <button
      aria-label="reload"
      class="counter-button reload"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-arrow-rotate-right "
        data-icon="arrow-rotate-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M386.3 160H336c-17.7 0-32 14.3-32 32s14.3 32 32 32H464c17.7 0 32-14.3 32-32V64c0-17.7-14.3-32-32-32s-32 14.3-32 32v51.2L414.4 97.6c-87.5-87.5-229.3-87.5-316.8 0s-87.5 229.3 0 316.8s229.3 87.5 316.8 0c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0c-62.5 62.5-163.8 62.5-226.3 0s-62.5-163.8 0-226.3s163.8-62.5 226.3 0L386.3 160z"
          fill="currentColor"
        />
      </svg>
    </button>
    <div
      class="counter-container"
    >
      <button
        aria-label="subtract"
        class="counter-button subtract"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-minus "
          data-icon="minus"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M432 256c0 17.7-14.3 32-32 32L48 288c-17.7 0-32-14.3-32-32s14.3-32 32-32l352 0c17.7 0 32 14.3 32 32z"
            fill="currentColor"
          />
        </svg>
      </button>
      <div
        class="flipper-container"
      >
        <div
          class="counter "
        >
          <span>
            0
          </span>
        </div>
      </div>
      <button
        aria-label="add"
        class="counter-button add"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-plus "
          data-icon="plus"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
